### PR TITLE
Add categories api

### LINKED ISSUE #8: Add api for fetching categories
API: "/categories"
Get all categories with their associated tags sorted by name.

### PR DESCRIPTION
**Pull request by SM43**
Get all categories with their tags sorted by name
Closes #8

### CODE REVIEW

File: api/.env.dev [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+POSTGRESQL_HOST="localhost"
+POSTGRESQL_PORT="5432"
+POSTGRESQL_DATABASE="tekton_hub"
+POSTGRESQL_USER="postgres"
+POSTGRESQL_PASSWORD="postgres"
+
+GITHUB_TOKEN=""
+CLIENT_ID=""
+CLIENT_SECRET=""
+JWT_SIGNING_KEY=""
```
Comment (sthaha) [inline api/.env.dev:12]: are these relevant to this commit?

File: api/cmd/hub/http.go [hunk 2]
```diff
@@ -103,10 +102,10 @@ func handleHTTPServer(ctx context.Context, u *url.URL, apiEndpoints *api.Endpoin
 // errorHandler returns a function that writes and logs the given error.
 // The function also writes and logs the error unique ID so that it's possible
 // to correlate.
-func errorHandler(logger *log.Logger) func(context.Context, http.ResponseWriter, error) {
+func errorHandler(logger *zap.SugaredLogger) func(context.Context, http.ResponseWriter, error) {
 	return func(ctx context.Context, w http.ResponseWriter, err error) {
 		id := ctx.Value(middleware.RequestIDKey).(string)
 		w.Write([]byte("[" + id + "] encoding: " + err.Error()))
-		logger.Printf("[%s] ERROR: %s", id, err.Error())
+		logger.Infof("[%s] ERROR: %s", id, err.Error())
```
Comment (sthaha) [inline api/cmd/hub/http.go:109]: Warn?

File: api/cmd/hub/main.go [hunk 3]
```diff
@@ -26,30 +27,36 @@ func main() {
...
 	}
 
 	// Initialize the services.
 	var (
-		apiSvc api.Service
+		categorySvc category.Service
 	)
 	{
-		apiSvc = hub.NewAPI(logger)
+		categorySvc = hub.NewCategory(api.DB(), logger)
```
Comment (sthaha) [inline api/cmd/hub/main.go:50]: I think it is better to pass then entire `api` that has both `db` and `logger`

File: api/pkg/app/app.go [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+		db.Host, db.Port, db.User, db.Name)
+}
+
+func (db *Database) ConnectionString() string {
+	return fmt.Sprintf(
+		"host=%s port=%s user=%s password=%s dbname=%s sslmode=disable",
+		db.Host, db.Port, db.User, db.Password, db.Name)
+}
+
+type GitHub struct {
```
Comment (sthaha) [inline api/pkg/app/app.go:62]: Is this needed for this patch?

File: api/design/api.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	Version("1.0")
+	Meta("swagger:example", "false")
+	Server("hub", func() {
+		Services("category", "swagger")
+		Host("localhost", func() {
+			URI("http://localhost:8000")
+		})
+	})
+	cors.Origin("/.*localhost/", func() {
+		cors.Headers("X-Shared-Secret")
```
Comment (PuneetPunamiya) [inline api/design/api.go:20]: Do we need to modify the headers?

**Comment by SM43** [comment]
Restructured in #11
